code/modules/antagonists/traitor/objectives/assassination.dm

```diff
@@ -0,0 +1,248 @@
+	 */
+	var/heads_of_staff = FALSE
+	///target we need to kill
+	var/mob/living/kill_target
+
+/datum/traitor_objective/assassinate/calling_card
+	name = "Assassinate \[TARGET] the \[JOB TITLE], and plant a calling card"
+	description = "Kill your target and plant a calling card in the pockets of your victim. If your calling card gets destroyed before you are able to plant it, this objective will fail."
+
+	var/obj/item/paper/calling_card/card
+
+/datum/traitor_objective/assassinate/calling_card/heads_of_staff
+	progression_reward = list(7 MINUTES, 10 MINUTES)
+	telecrystal_reward = list(2, 3)
+
+	heads_of_staff = TRUE
+
+/datum/traitor_objective/assassinate/behead
+	name = "Behead \[TARGET], the \[JOB TITLE]"
+	description = "Behead and hold \[TARGET]'s head to succeed this objective. If the head gets destroyed before you can do this, you will fail this objective."
+
+	///the body who needs to hold the head
+	var/mob/living/needs_to_hold_head
+	///the head that needs to be picked up
+	var/obj/item/bodypart/head/behead_goal
+
+/datum/traitor_objective/assassinate/behead/heads_of_staff
+	progression_reward = list(7 MINUTES, 10 MINUTES)
+	telecrystal_reward = list(2, 3)
+
+	heads_of_staff = TRUE
+
+
+/datum/traitor_objective/assassinate/calling_card/generate_ui_buttons(mob/user)
+	var/list/buttons = list()
+	if(!card)
+		buttons += add_ui_button("", "Pressing this will materialize a calling card, which you must plant to succeed.", "paper-plane", "summon_card")
+	return buttons
+
+/datum/traitor_objective/assassinate/calling_card/ui_perform_action(mob/living/user, action)
+	. = ..()
+	switch(action)
+		if("summon_card")
+			if(card)
+				return
+			card = new(user.drop_location())
+			user.put_in_hands(card)
+			card.balloon_alert(user, "the card materializes in your hand")
+			RegisterSignal(card, COMSIG_ITEM_EQUIPPED, .proc/on_card_planted)
+			AddComponent(/datum/component/traitor_objective_register, card, \
+				succeed_signals = null, \
+				fail_signals = COMSIG_PARENT_QDELETING, \
+				penalty = TRUE)
+
+/datum/traitor_objective/assassinate/calling_card/proc/on_card_planted(datum/source, mob/living/equipper, slot)
+	SIGNAL_HANDLER
+	if(equipper != kill_target)
+		return //your target please
+	if(equipper.stat != DEAD)
+		return //kill them please
+	if(slot != ITEM_SLOT_LPOCKET && slot != ITEM_SLOT_RPOCKET)
+		return //in their pockets please
+	succeed_objective()
+
+/datum/traitor_objective/assassinate/calling_card/generate_objective(datum/mind/generating_for, list/possible_duplicates)
+	. = ..()
+	if(!.) //didn't generate
+		return FALSE
+	RegisterSignal(kill_target, COMSIG_PARENT_QDELETING, .proc/on_target_qdeleted)
+
+/datum/traitor_objective/assassinate/calling_card/ungenerate_objective()
+	UnregisterSignal(kill_target, COMSIG_PARENT_QDELETING)
+	. = ..() //unsets kill target
+	if(card)
+		UnregisterSignal(card, COMSIG_ITEM_EQUIPPED)
+	card = null
+
+/datum/traitor_objective/assassinate/calling_card/on_target_qdeleted()
```

Comment:
SIGNAL_HANDLER

Reply:
you can't put SIGNAL_HANDLER on overriden procs, it has to go on the base proc, which it is.